pandas/core/resample.py

```diff
@@ -404,7 +404,10 @@ def _wrap_result(self, result):
 
         if isinstance(result, ABCSeries) and result.empty:
             obj = self.obj
-            result.index = obj.index._shallow_copy(freq=to_offset(self.freq))
+            if isinstance(obj.index, PeriodIndex):
+                result.index = obj.index.asfreq(self.freq)
```

Comment:
hmm, maybe should have ``.asfreq()`` on the other indices?

Reply:
I briefly thought about that. I *think* it'll only be useful for going from a DTI without and `freq` to a DTI with a freq. Changing from one freq to another doesn't really make sense for DTI or TDI.

Reply:
@jbrockmendel thoughts here?

Reply:
It isn't immediately obvious what DTI.asfreq means.  Is the motivation just to avoid the need for checks like this?

Reply:
yes

Reply:
Definitely skeptical of making a public method that is that ambiguous.  If this pattern is repeated a lot then a private method to this effect might be an easier sell.

Reply:
Given that there's only 1 place in pandas, (and 2 in the tests), I'd prefer to leave this as is rather than define a new method.

Reply:
yeah ok, just trying to avoid special cases.